### PR TITLE
Move vendor to fbjs package

### PR DESCRIPTION
**Pull request by zpao**
Ultimately we'll ship `fbjs` as a standalone package but for now, I'm just going to split it out and build it here. The result of keeping it here actually means we're in a bit of a hackier state but move fast.

## TODO
- [ ] ~~Make `grunt test` work (especially since it's all that's running right on Travis)~~
- [x] ship to npm
- [ ] ~~look into all the other testing (sauce)~~
- [x] update for fbjs version

### CODE REVIEW

**Comment by zpao** [comment]
```
   raw     gz Compared to last run
     =      = build/JSXTransformer.js
  +505    -19 build/react-with-addons.js
   +27    -43 build/react-with-addons.min.js
  +490    -36 build/react.js
   +38     -4 build/react.min.js
```

And we're back in business. Before the change to browserifying we were not envifying the code coming from fbjs. The result was that we left `process.env.NODE_ENV` in there, which then also resulted in the `process` shim being bundled. We could have solved this by specifying the envify transform over there (and maybe we should) but the upshot of this is that we use the env we want.

I think this is pretty much good to go. We'll stabilize fbjs a bit more and bump versions before final.

**Comment by zpao** [comment]
Lint failure is https://github.com/babel/babel-eslint/issues/149

File: scripts/jest/preprocessor.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+var ts = tsPreprocessor(defaultLibraries);
+
+// This assumes the module map has been built. This might not be safe.
+// We should consider consuming this from a built fbjs module from npm.
+var moduleMap = require('fbjs/module-map');
+var babelPluginDEV = require('fbjs/scripts/babel/dev-expression');
+var babelPluginRequires = require('fbjs/scripts/babel/rewrite-requires');
+
+
+// console.warn(moduleMap);
```
Comment (spicyj) [inline scripts/jest/preprocessor.js]: remove

File: scripts/jest/preprocessor.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if (path.match(/\.coffee$/)) {
+      return coffee.compile(src, {'bare': true});
+    }
+    if (path.match(/\.ts$/) && !path.match(/\.d\.ts$/)) {
+      return ts.compile(src, path);
+    }
+    // TODO: make sure this stays in sync with gulpfile
+    if (!path.match(/\/node_modules\//) && !path.match(/\/third_party\//)) {
+      if (/__tests__/.test(path)) {
+        // console.warn(src);
```
Comment (spicyj) [inline scripts/jest/preprocessor.js]: remove

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+gulp.task('react:clean', function(cb) {
+  del([paths.react.lib], cb);
+});
+
+gulp.task('react:modules', function() {
+  return gulp
+    .src(paths.react.src)
+    .pipe(babel(babelOpts))
```
Comment (spicyj) [inline gulpfile.js]: Does this cache somehow? Seems to take ~3s on my computer which feels weirdly fast.

File: Gruntfile.js [hunk 4]
```diff
@@ -21,6 +23,17 @@ module.exports = function(grunt) {
 
   grunt.config.set('compress', require('./grunt/config/compress'));
 
+  function spawnGulp(args, opts, done) {
+    grunt.util.spawn({
+      cmd: 'gulp',
```
Comment (spicyj) [inline Gruntfile.js]: Can this be node_modules/gulp/bin/gulp.js? I don't have gulp installed globally.

File: Gruntfile.js [hunk 5]
```diff
@@ -21,6 +23,17 @@ module.exports = function(grunt) {
...
+      args: args,
+      opts: assign({stdio: 'inherit'}, opts),
+    }, function() {
+      // TODO maybe add handling of status codes
+      done();
```
Comment (spicyj) [inline Gruntfile.js]: Status code handling plz. This failed silently on my machine due to the above. ^

File: gulpfile.js (hunk 3 above)
Comment (zpao) [inline gulpfile.js]: I don't think so. My `~/.babel.json` doesn't get touched so I'm pretty sure that's not doing something we're not expecting.

File: Gruntfile.js (hunk 5 above)
Comment (zpao) [inline Gruntfile.js]: :thumbsup:

**Comment by zpao** [comment]
Shipping. We'll update to a newer version of fbjs soon.

File: gulpfile.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    src: [
+      'src/**/*.js',
+      '!src/**/__tests__/**/*.js',
+      '!src/**/__mocks__/**/*.js',
+    ],
+    lib: 'build/modules',
+  },
+};
+
+var babelOpts = {
```
Comment (bgw) [inline gulpfile.js]: I think this should be in a .babelrc file (which gulp-babel reads). You might be able to better share the configuration between this and jest that way.

Comment (zpao) [inline gulpfile.js]: Way ahead of you. There are reasons it's not and we'll work them out.

**Pull request merged**